guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpPostHelper.java

```diff
@@ -208,9 +207,13 @@ private String uploadPom( final FormData formData ) throws IOException {
                                    version );
                 }
             } catch (ProjectBuildingException pbe) {
-                return pbe.getMessage();
+                log.error(pbe.getMessage(),
+                          pbe);
+                return UPLOAD_UNABLE_TO_PARSE_POM;
             } catch (MavenEmbedderException mee) {
-                return mee.getMessage();
+                log.error(mee.getMessage(),
+                          mee);
+                return UPLOAD_UNABLE_TO_PARSE_POM;
             }catch (Exception e) {
                 return UPLOAD_UNABLE_TO_PARSE_POM;
             }
```

Comment:
This is used by the file upload servlet so the UncaughtExceptionHandler might not pick it up, but true we should probably just capture any exception.